pkg/operator/controller/route-status/controller_test.go

```diff
@@ -0,0 +1,199 @@
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "not-shard-label", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "shard-label", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", ""),
+		},
+		{
+			name:              "admitted with no routeSelector and expression namespaceSelector with route with correct labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", "shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", "shard-label"),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "not admitted with no routeSelector and expression namespaceSelector with route with incorrect labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", "not-shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", "shard-label"),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", ""),
+		},
+		{
+			name:              "route is nil",
+			request:           reconcileRequestRoute,
+			route:             nil,
+			namespace:         newNamespace("ns", "not-shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", "shard-label"),
+			expectedErr:       false,
+		},
+		{
+			name:              "admitted by ingress controller that doesn't exist anymore",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "missing-ic"),
+			namespace:         newNamespace("ns", "shard-label"),
+			ingressController: nil,
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "missing-ic"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			r := newFakeReconciler(tc.namespace)
+			if tc.route != nil {
+				if err := r.client.Create(context.Background(), tc.route); err != nil {
+					t.Errorf("error creating route: %v", err)
+				}
+			}
+			if tc.ingressController != nil {
+				if err := r.client.Create(context.Background(), tc.ingressController); err != nil {
+					t.Errorf("error creating ingress controller: %v", err)
+				}
+			}
+
+			if _, err := r.Reconcile(context.Background(), tc.request); tc.expectedErr {
+				if err == nil {
+					t.Errorf("expected error, got no error")
+				}
```

Comment:
```suggestion
			if _, err := r.Reconcile(context.Background(), tc.request); tc.expectedErr && err == nil {
				t.Errorf("expected error, got no error")
```
You could flatten this if condition without affecting the test functionality

Reply:
Actually, it would slightly change the test functionality. If `tc.expectedErr` is `true` and `err` is not `nil` (which is a successful/passing test), this logic flow says to not check against the expectedRoute (aka the error check is enough).

However, this doesn't matter because I don't use the `tc.expectedErr`. I felt it nice to have it there for future unit test capabilities. I modified it to flatten and check `} else if tc.expectedRoute != nil && !tc.expectedErr {` down in the last if/else case.